src/Error/TypeError.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Error;
+
+use Exception;
+
+/**
+ * Caused if a user passes a wrong type.
+ */
+class TypeError extends Exception implements ClientAware
```

Comment:
Not sure if we should use `UserError` instead or something else entirely. I can not use `Error`, since it requires passing a message, which is explictly *not* done.

Reply:
Good point. We should also preserve `graphql` category for those kinds of errors as they are described in the GraphQL spec: https://spec.graphql.org/draft/#sec-Values-of-Correct-Type (and also for backwards compatibility)

So maybe just make `message` in the `Error` optional and use it instead?

P.S. This whole piece of catching and wrapping is confusing though. Maybe worth a refactor someday.